package/MDAnalysis/core/topologyattrs.py

```diff
@@ -2041,6 +2051,412 @@ def total_charge(group, compound='group'):
+        numpy.ndarray
+            Dipole moment(s) of (compounds of) the group in :math:`eÅ`.
+            If `compound` was set to ``'group'``, the output will be a single
+            value. Otherwise, the output will be a 1d array of shape ``(n,)``
+            where ``n`` is the number of compounds.
+
+
+        .. versionadded:: 2.4.0
+        """
+
+        atomgroup = group.atoms
+
+        dipole_vector = atomgroup.dipole_vector(**kwargs)
+
+        if len(dipole_vector.shape) > 1:
+            dipole_moment = np.sqrt(np.sum(dipole_vector**2, axis=1))
+        else:
+            dipole_moment = np.sqrt(np.sum(dipole_vector**2))
+
+        return dipole_moment
+
+    transplants[GroupBase].append(
+        ('dipole_moment', dipole_moment))
+
+    @warn_if_not_unique
+    @_pbc_to_wrap
+    @check_wrap_and_unwrap
+    def quadrupole_tensor(group, wrap=False, unwrap=False, 
+        compound='group', center="mass"):
+        r"""Traceless quadrupole tensor of the group or compounds.
+
+        .. math::
+            \mathsf{Q} = \sum_{i=1}^{N} q_{i} ( \mathbf{r_{i}} - 
+            \mathbf{r_{COM}} ) \cdot ( \mathbf{r_{i}} - \mathbf{r_{COM}} )
+
+        .. math::
+            \hat{\mathsf{Q}} = \frac{3}{2} \mathsf{Q} - \frac{1}{2} 
+            tr(\mathsf{Q})
+
+        where :math:`\hat{\mathsf{Q}}` is the traceless quadrupole 
+        tensor.
+         
+        Computes the quadrupole tensor of :class:`Atoms<Atom>` in the group.
+        Tensor per :class:`Residue`, :class:`Segment`, molecule, or
+        fragment can be obtained by setting the `compound` parameter
+        accordingly.
+
+        Note that when there is an unsymmetrical plane in the molecule or 
+        group, the magnitude of the quadrupole tensor is dependent on the 
+        ``center`` (e.g., :math:`r_{COM}`) chosen and cannot be translated.
+
+        Parameters
+        ----------
+        group : obj
+            AtomGroup
+        wrap : bool, optional
+            If ``True`` and `compound` is ``'group'``, move all atoms to the
+            primary unit cell before calculation.
+            If ``True`` and `compound` is not ``group``, the
+            centers of mass of each compound will be calculated without moving
+            any atoms to keep the compounds intact.
+        unwrap : bool, optional
+            If ``True``, compounds will be unwrapped before computing their
+            centers.
+        compound : {'group', 'segments', 'residues', 'molecules', \
+                    'fragments'}, optional
+            If ``'group'``, a single quadrupole value returns. Otherwise, an
+            array of each :class:`Segment`, :class:`Residue`, molecule, or
+            fragment will be returned as an array of position vectors, i.e.
+            a 1d array.
+            Note that, in any case, *only* the positions of
+            :class:`Atoms<Atom>` *belonging to the group* will be taken into
+            account.
+        center : str, optional
```

Comment:
Can be explicit with choices:
```suggestion
        center : {'mass', 'charge'}, optional
```